  <main role="main">    <h1>Class InvocationTargetException</h1>   <ul class="inheritance"> <li><a href="../object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="../reflectiveoperationexception">java.lang.ReflectiveOperationException</a></li> <li> <ul class="inheritance"> <li>java.lang.reflect.InvocationTargetException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><code><a href="../../io/serializable">Serializable</a></code></dd> </dl>  <pre data-language="java">public class InvocationTargetException
extends ReflectiveOperationException</pre> <div>
<p>InvocationTargetException is a checked exception that wraps an exception thrown by an invoked method or constructor. </p>
<p>As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "target exception" that is provided at construction time and accessed via the <a href="#getTargetException()"><code>getTargetException()</code></a> method is now known as the <i>cause</i>, and may be accessed via the <a href="../throwable#getCause()"><code>Throwable.getCause()</code></a> method, as well as the aforementioned "legacy method."</p>
</div> <dl> <dt>Since:</dt> <dd>1.1</dd> <dt>See Also:</dt> <dd>
<a href="method"><code>Method</code></a>, <a href="constructor"><code>Constructor</code></a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/serialized-form.html#java.lang.reflect.InvocationTargetException">Serialized Form</a>
</dd> </dl>      <section role="region">  <h2 id="constructor.summary">Constructors</h2> <table>  <tr> <th scope="col">Modifier</th> <th scope="col">Constructor</th> <th scope="col">Description</th> </tr> <tr> <td><code>protected </code></td> <th scope="row"><code><a href="#%3Cinit%3E()">InvocationTargetException</a>()</code></th> <td> <p>Constructs an <code>InvocationTargetException</code> with <code>null</code> as the target exception.</p> </td> </tr> <tr> <td></td> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.Throwable)">InvocationTargetException</a>​(<a href="../throwable">Throwable</a> target)</code></th> <td> <p>Constructs a InvocationTargetException with a target exception.</p> </td> </tr> <tr> <td></td> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.Throwable,java.lang.String)">InvocationTargetException</a>​(<a href="../throwable">Throwable</a> target,
                         <a href="../string">String</a> s)</code></th> <td> <p>Constructs a InvocationTargetException with a target exception and a detail message.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="method.summary">Methods</h2> <table>  <tr> <th scope="col">Modifier and Type</th> <th scope="col">Method</th> <th scope="col">Description</th> </tr> <tr id="i0"> <td><code><a href="../throwable">Throwable</a></code></td> <th scope="row"><code><a href="#getCause()">getCause</a>()</code></th> <td> <p>Returns the cause of this exception (the thrown target exception, which may be <code>null</code>).</p> </td> </tr> <tr id="i1"> <td><code><a href="../throwable">Throwable</a></code></td> <th scope="row"><code><a href="#getTargetException()">getTargetException</a>()</code></th> <td> <p>Get the thrown target exception.</p> </td> </tr> </table>  <h2 id="methods.inherited.from.class.java.lang.Throwable">Methods declared in class java.lang.<a href="../throwable">Throwable</a>
</h2> <code><a href="../throwable#addSuppressed(java.lang.Throwable)">addSuppressed</a>, <a href="../throwable#fillInStackTrace()">fillInStackTrace</a>, <a href="../throwable#getLocalizedMessage()">getLocalizedMessage</a>, <a href="../throwable#getMessage()">getMessage</a>, <a href="../throwable#getStackTrace()">getStackTrace</a>, <a href="../throwable#getSuppressed()">getSuppressed</a>, <a href="../throwable#initCause(java.lang.Throwable)">initCause</a>, <a href="../throwable#printStackTrace()">printStackTrace</a>, <a href="../throwable#printStackTrace(java.io.PrintStream)">printStackTrace</a>, <a href="../throwable#printStackTrace(java.io.PrintWriter)">printStackTrace</a>, <a href="../throwable#setStackTrace(java.lang.StackTraceElement%5B%5D)">setStackTrace</a>, <a href="../throwable#toString()">toString</a></code>  <h2 id="methods.inherited.from.class.java.lang.Object">Methods declared in class java.lang.<a href="../object">Object</a>
</h2> <code><a href="../object#clone()">clone</a>, <a href="../object#equals(java.lang.Object)">equals</a>, <a href="../object#finalize()">finalize</a>, <a href="../object#getClass()">getClass</a>, <a href="../object#hashCode()">hashCode</a>, <a href="../object#notify()">notify</a>, <a href="../object#notifyAll()">notifyAll</a>, <a href="../object#wait()">wait</a>, <a href="../object#wait(long)">wait</a>, <a href="../object#wait(long,int)">wait</a></code>  </section>   <div>   <section role="region">  <h2 id="constructor.detail">Constructors</h2>   <h3 id="&lt;init&gt;()">InvocationTargetException</h3> <pre data-language="java">protected InvocationTargetException()</pre> <p>Constructs an <code>InvocationTargetException</code> with <code>null</code> as the target exception.</p>    <h3 id="&lt;init&gt;(java.lang.Throwable)">InvocationTargetException</h3> <pre data-language="java">public InvocationTargetException(Throwable target)</pre> <p>Constructs a InvocationTargetException with a target exception.</p> <dl> <dt>Parameters:</dt> <dd>
<code>target</code> - the target exception</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.Throwable,java.lang.String)">InvocationTargetException</h3> <pre data-language="java">public InvocationTargetException(Throwable target,
                                 String s)</pre> <p>Constructs a InvocationTargetException with a target exception and a detail message.</p> <dl> <dt>Parameters:</dt> <dd>
<code>target</code> - the target exception</dd> <dd>
<code>s</code> - the detail message</dd> </dl>   </section>  <section role="region">  <h2 id="method.detail">Methods</h2>   <h3 id="getTargetException()">getTargetException</h3> <pre data-language="java">public Throwable getTargetException()</pre> <div>
<p>Get the thrown target exception. </p>
<p>This method predates the general-purpose exception chaining facility. The <a href="../throwable#getCause()"><code>Throwable.getCause()</code></a> method is now the preferred means of obtaining this information.</p>
</div> <dl> <dt>Returns:</dt> <dd>the thrown target exception (cause of this exception).</dd> </dl>    <h3 id="getCause()">getCause</h3> <pre data-language="java">public Throwable getCause()</pre> <p>Returns the cause of this exception (the thrown target exception, which may be <code>null</code>).</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../throwable#getCause()">getCause</a></code> in class <code><a href="../throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the cause of this exception.</dd> <dt>Since:</dt> <dd>1.4</dd> </dl>   </section>  </div>  </main>  <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993, 2020, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/InvocationTargetException.html" class="_attribution-link">https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/InvocationTargetException.html</a>
  </p>
</div>
